js/derivadex.js

```diff
@@ -0,0 +1,599 @@
+        };
+        const orderBookResponse = await this.publicGetMarketsOrderBookL2Symbol (this.extend (request, params)); // markets/order_book endpoint response is cached for 10 seconds
+        const marketsValue = marketsResponse['value'][0];
+        const orderBookValue = orderBookResponse['value'];
+        const quoteVolume = this.safeString (marketsValue, 'volume');
+        const timestamp = this.safeString (marketsResponse, 'timestamp');
+        const close = this.safeString (marketsValue, 'price');
+        const bid = this.safeString (orderBookValue[0], 'price');
+        const bidVolume = this.safeString (orderBookValue[0], 'amount');
+        const ask = this.safeString (orderBookValue[1], 'price');
+        const askVolume = this.safeString (orderBookValue[1], 'amount');
+        return {
+            'symbol': symbol,
+            'timestamp': timestamp,
+            'datetime': undefined,
+            'high': undefined,
+            'low': undefined,
+            'bid': bid,
+            'bidVolume': bidVolume,
+            'ask': ask,
+            'askVolume': askVolume,
+            'vwap': undefined,
+            'open': undefined,
+            'close': close,
+            'last': close,
+            'previousClose': undefined,
+            'change': undefined,
+            'percentage': undefined,
+            'average': undefined,
+            'baseVolume': undefined,
+            'quoteVolume': quoteVolume,
+            'info': marketsResponse,
+        };
+    }
+
+    async fetchMyTrades (symbol = undefined, since = undefined, limit = undefined, params = {}) {
+        /**
+         * @method
+         * @name derivadex#fetchMyTrades
+         * @description fetch all trades made by the user
+         * @param {string|undefined} symbol unified market symbol
+         * @param {int|undefined} since the earliest time in ms to fetch trades for
+         * @param {int|undefined} limit the maximum number of trades structures to retrieve
+         * @param {object} params extra parameters specific to the derivadex api endpoint
+         * @returns {[object]} a list of [trade structures]{@link https://docs.ccxt.com/en/latest/manual.html#trade-structure}
+         */
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        const request = {
+            'symbol': market['id'],
+        };
+        if (limit !== undefined) {
+            request['limit'] = limit; // default 500
+        }
+        const response = await this.publicGetFills (this.extend (request, params));
+        response['traderAddress'] = ''; // TODO: supply the users trader address
+        return await this.parseTrades (response, market, since, limit);
+    }
+
+    async fetchTrades (symbol, since = undefined, limit = undefined, params = {}) {
+        /**
+         * @method
+         * @name derivadex#fetchTrades
+         * @description get the list of most recent trades for a particular symbol
```

Comment:
if the purpose of this API is to get the most recent trades for a particular symbol then Im not sure that our current paging mechanism works since it returns rows starting from the beginning of time